src/tests/test_certstor_linux.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+* (C) 1999-2019 Jack Lloyd
+* (C) 2019      René Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_CERTSTOR_FLATFILE) && defined(BOTAN_SYSTEM_CERT_BUNDLE)
+
+#include "test_certstor_utils.h"
+#include <botan/certstor_flatfile.h>
+#include <botan/ber_dec.h>
+#include <botan/der_enc.h>
+#include <botan/hex.h>
+
+namespace Botan_Tests {
+
+namespace {
+
+Test::Result open_certificate_store()
+   {
+   Test::Result result("linux Certificate Store - Open Store");
+
+   try
+      {
+      result.start_timer();
+      Botan::Flatfile_Certificate_Store unused(BOTAN_SYSTEM_CERT_BUNDLE);
+      result.end_timer();
+      result.test_gt("found some certificates", unused.all_subjects().size(), 0);
```

Comment:
I'd say you don't want to depend on a well-formed system cert store for the unit tests. Also this would make the test for `Flatfile_Certificate_Store` dependent on Linux, right?

Instead, I'd suggest to put together a small PEM bundle (best case with certificates that are valid for as many years as possible and of course the certificates we are expecting to find + a few more) and put it in the `src/tests/data/x509/misc`. Here you can can access the file path via:

```
Test::data_file("x509/misc/valid_ca_bundle.pem")
```

Same is true for the coming tests, obviously.

Reply:
This would be fine.

During cert verification it is possible to set the reference time, so cert expiration doesn't matter. The existing path validation tests do this.

Reply:
We do want some way of checking that the actual system cert store works (for example to find out about issues like #1890) but that can be done by extending the command line utility with subprograms to print the list of CAs in the store, verify using the store, etc.

Reply:
I added two bundles:
* `valid_ca_bundle.pem` containing:
  * D-TRUST Root Class 3 CA 2 EV 2009
  * DST Root CA X3
  * VeriSign Class 3 Public Primary Certification Authority - G3
    (a X.509 V1 certificate)
* `ca_bundle_containing_non_ca.pem` containing:
  * DST Root CA X3
  * some old eduroam certificate of mine
    (obviously not a CA)